be/src/exec/vectorized/hash_join_node.cpp

```diff
@@ -574,6 +561,28 @@ static inline bool check_chunk_zero_and_create_new(ChunkPtr* chunk) {
     return false;
 }
 
+void HashJoinNode::_evaluate_build_keys(const ChunkPtr& chunk) {
+    _key_columns.resize(0);
+    size_t num_rows = chunk->num_rows();
+    for (auto& ctx : _build_expr_ctxs) {
+        const TypeDescriptor& data_type = ctx->root()->type();
+        ColumnPtr key_column = ctx->evaluate(chunk.get());
+        if (key_column->is_constant()) {
```

Comment:
suggest to classify column into 4 types:
1. only_null(), const null column,
2. is_constant(),  constant column,
3. is_nullable(), nullable column
4. otherwise,  plain simple column.

Reply:
Fixed